### PR TITLE
Respect locked utxos

### PR DESCRIPTION
**Pull request by dannydeezy**
Locked utxos should not be available to spend. (https://github.com/ordinals/ord/issues/2496)

This PR attempts to ensure that if a utxo has been locked with the `lockunspent` bitcoin RPC call, then it will not be selected as a cardinal utxo. This enables people to safely lock utxos that they don't want ord to spend when doing inscriptions or sends (unless they are deliberately inscribing/sending that particular utxo)

### CODE REVIEW

**Comment by dannydeezy** [comment]
Todo:
- [ ] Fix merge conflicts
- [ ] Write tests

File: src/index.rs [hunk 1]
```diff
@@ -315,19 +333,13 @@ impl Index {
         }),
     );
 
-    #[derive(Deserialize)]
```
Comment (dannydeezy) [inline src/index.rs:318]: this logic has been extracted into the above `get_locked_outputs` function

**Comment by dannydeezy** [comment]
@casey @raphjaph I believe this is ready for review

**Comment by raphjaph** [comment]
Could you allow edits to this PR by maintainers?  I'd like to rebase/merge in newest changes before reviewing.

**Comment by raphjaph** [comment]
I still can't push things unfortunately. There's a linting test that isn't passing. Could you try closing this PR and reopening a new one? Maybe it'll update the permissions then

**Comment by dannydeezy** [comment]
opened new one: https://github.com/ordinals/ord/pull/2563

**Pull request closed without merging**